ProvinceMapper/Source/Frames/Images/ImageCanvas.cpp

```diff
@@ -109,6 +109,45 @@ void ImageCanvas::strafeProvince(const std::shared_ptr<Province>& province)
 			strafedPixels.emplace_back(pixel);
 }
 
+void ImageCanvas::highlightRegionByCommentRow(const int row)
+{
+	if (!activeVersion)
+	{
+		return;
+	}
+	const auto& links = activeVersion->getLinks();
+	if (row >= static_cast<int>(links->size()))
+	{
+		return;
+	}
+	if (const auto& commentLink = links->at(row); commentLink->getComment())
+	{
+		auto rowUnderComment = row + 1;
+		while (static_cast<unsigned long>(rowUnderComment) < links->size() && !links->at(rowUnderComment)->getComment())
+		{
+			const auto& link = links->at(rowUnderComment);
+			// Highlight our provinces' pixels.
+			highlightProvinces(link);
+			++rowUnderComment;
+		}
+	}
+}
+
+void ImageCanvas::highlightProvinces(const std::shared_ptr<LinkMapping>& linkToHighlight)
+{
+	for (const auto& province: getRelevantProvinces(linkToHighlight))
+		highlightProvince(province);
+}
+
+void ImageCanvas::highlightProvince(const std::shared_ptr<Province>& province)
+{
+	for (const auto& pixel: province->innerPixels)
+	{
+		highlightedProvinces.emplace_back(province);
```

Comment:
You can move this outside the loop.

Reply:
Thanks! This was the reason why dehighlighting was slower than highlighting.